File: 682-baseball-game/682-baseball-game.cpp
class Solution {
public:
    int calPoints(vector<string>& ops) {
        vector<int>stk ;
        for(string x:ops){
            int n = stk.size() ;
            if(isdigit(x[x.length()-1])){
                stk.push_back(stoi(x));
            }else if(x=="C")stk.pop_back() ;
            else if(x=="+"){
                stk.push_back(stk[n-1]+stk[n-2]);
            }else{
                stk.push_back(stk[n-1]*2);
            }
        }
        return accumulate(stk.begin(),stk.end(),0);
    }
};

File: 682-baseball-game/682-baseball-game.java
class Solution {
public int calPoints(String[] ops) {
    int ans = 0;
    Stack<Integer> stk = new Stack<>();
    for (String op : ops) {
        if (op.equals("C")) { ans -= stk.pop (); continue; }
        else if (op.equals ("D")) stk.push (stk.peek () * 2);
        else if (op.equals ("+")) stk.push (stk.get(stk.size () - 1) + stk.get(stk.size () - 2));
        else stk.push (Integer.valueOf (op));
        ans += stk.peek(); 
    }
    return ans;
}
}